pkg/util/util.go

```diff
@@ -144,3 +145,16 @@ func IsReadOnlyDisk(disk *v1.Disk) bool {
 
 	return isReadOnlyCDRom
 }
+
+func AlignImageSize(size int64, logger *log.FilteredLogger) int64 {
+	remainder := size % 4096
+	if remainder == 0 {
+		return size
+	} else {
+		newSize := size - remainder
```

Comment:
Why reduce the size and not increase it?

Reply:
The function is used to create valid images inside given volumes of size `size`.
In most cases, creating an image bigger than that would just fail for lack of space.

Reply:
Got you. Thanks for the clarification!

Reply:
Might want to align on 1Mi or 4Mi since there are storage devices out there with larger block sizes (Azure blob comes to mind). Just an FYI, CDI aligns it disks on 1Mi boundaries when resizing.

Reply:
Having more thoughts on this. CDI already exposes some library functions, would it make sense for CDI to expose the function that calculates the alignment so KubeVirt doesn't have to re-implement it?

Reply:
This needs to work even when CDI is not present.
What we need to satisfy here is the libvirt requirement for 4k alignment. Platform-related constraints should probably be dealt with separately.

Reply:
The actual requirement is that the image aligns with the hardware block size of the underlying storage. Which in MOST cases is either 512bytes or 4k, but we are seeing 1Mi and 4Mi popping up as well. This is the reason CDI now aligns on 1Mi instead.

Reply:
Thanks for aligning to 1Mi.